### PR TITLE
[W5.1a][W12-1] Lim Jit Wei

### PR DESCRIPTION
**Pull request by jitwei98**
Delegate methods that are private and relevant to the formatting of the output of TextUi to a new class Formatter

### CODE REVIEW

File: src/seedu/addressbook/ui/TextUi.java [hunk 1]
```diff
@@ -41,6 +37,7 @@
 
     private final Scanner in;
     private final PrintStream out;
+    private final Formatter formatter = new Formatter(this);
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/TextUi.java:40]: This is not really necessary. The `TextUI` can receive formatting related stuff and do the I/O

File: src/seedu/addressbook/formatter/Formatter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        for (ReadOnlyPerson person : persons) {
+            formattedPersons.add(person.getAsTextHidePrivate());
+        }
+        showToUserAsIndexedList(formattedPersons);
+    }
+
+    /**
+     * Shows a list of strings to the user, formatted as an indexed list.
+     */
+    public void showToUserAsIndexedList(List<String> list) {
```
Comment (okkhoy) [inline src/seedu/addressbook/formatter/Formatter.java:37]: This kinda increases the coupling (Heads up! coupling will be covered later and you will realize what I mean!) 
Also it violates SRP, think how?
A better approach is to do only formatting related stuff here and return the formatted string to the caller.

**Review by okkhoy — COMMENTED** [review]
You seem to have understood most part of SRP, but still a bit blur. Please review the relevant material in the text.

**Comment by okkhoy** [comment]
@jitwei98 
Some comments added. Please take note and close the pull request.

**Comment by jitwei98** [comment]
Noted with thanks !

**Pull request closed without merging**